### ISSUE TITLE
Unit test runner is failing with "System.InvalidOperationException: Collection was modified; enumeration operation may not execute."

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#10718]
http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/6344/console

```
17:14:58 EXEC : error : Unable to run Binaries\Debug\Roslyn.Services.Editor.UnitTests2.dll with C:\Users\dotnet-bot\.nuget\packages\xunit.runner.console\2.1.0\tools\xunit.console.x86.exe. System.InvalidOperationException: Collection was modified; enumeration operation may not execute. [D:\j\workspace\roslyn_prtest---345e02d2\BuildAndTest.proj]
17:14:58      at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
17:14:58      at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
17:14:58      at System.Collections.Generic.List`1.Enumerator.MoveNext()
17:14:58      at System.String.Join(String separator, IEnumerable`1 values)
17:14:58      at RunTests.ProcessTestExecutor.<RunTestAsync>d__7.MoveNext() in q:\roslyn\src\Tools\Source\RunTests\ProcessTestExecutor.cs:line 124
```

Only hit this once, @jaredpar , does this look familiar?

**Comment by davkean** [comment]
It looks like we're receiving a OutputDataReceived or ErrorDataReceived event after the Exited event has fired.

**Comment by jaredpar** [comment]
Never seen this before.  Looking.

**Comment by jaredpar** [comment]
According to docs I need to call `WaitForExit` to ensure the buffers are flushed.  Apparentyl listening to the Exit event isn't enough :frowning:

**Comment by davkean** [comment]
I walked through the code on sourceof.net, from what I could see, the event looked good enough - but there were three places it was fired.

**Comment by jasonmalinowski** [comment]
This was fixed in #10958.
